src/main/java/com/klin/holoItems/collections/gen0/suiseiCollection/items/Comet.java

```diff
@@ -73,113 +76,152 @@ public void registerRecipes(){
+        if(player.hasPotionEffect(PotionEffectType.WEAKNESS)){
+            player.sendMessage("§7The axe weighs heavily on your arms");
+            return;
+        }
+
+        // Cancel if on cooldown
+        ItemStack item = event.getItem();
+        if(Utility.onCooldown(item))
+            return;
+        Utility.cooldown(item, 20);
+
+        double damage = 4 + 3 * (Utility.checkPotionEffect(player, PotionEffectType.STRENGTH));
+
+        Location location = player.getEyeLocation();
+        World world = player.getWorld();
+
+        final double maxDistance = 50;
+        double distance = maxDistance;
+
+        // Normalized vector
+        Vector direction = location.getDirection();
+
+        // Set vector speed as 3 blocks/sec
+        final double speed = 3;
+        Vector dir = direction.multiply(speed);
+
+        Set<LivingEntity> targets = new HashSet<>();
+
+        // Raytrace to find entities in the way. If piercing is applied, do it multiple times
+        for(int i=0; i<1+item.getEnchantmentLevel(Enchantment.PIERCING); i++) {
+            RayTraceResult result = world.rayTrace(location, dir, maxDistance,
+                    FluidCollisionMode.NEVER, true, 0.5,
+                    entity -> (entity != player &&
+                            entity instanceof LivingEntity && !(entity instanceof ArmorStand) &&
+                            !targets.contains(entity))); // Skip previously raytraced entities
+            if (result != null) {
+                LivingEntity entity = (LivingEntity) result.getHitEntity();
+                if (entity != null) {
+                    distance = location.distance(result.getHitEntity().getLocation());
+                    targets.add(entity);
+                }
+                else if (result.getHitBlock() != null) {
+                    distance = location.distance(result.getHitBlock().getLocation());
+                    break;
+                }
+            }
+        }
+
+        // Offset axe according to player's hand
+        int yOffset;
+        if (event.getHand() == EquipmentSlot.HAND) {
+            yOffset = 1; //perpendicular to the right
+        } else {
+            yOffset = -1; //perpendicular to the left
+        }
+
+        Vector handOffset = direction.clone()
+            .crossProduct(new Vector(0, yOffset, 0)) // move perpendicular
+            .multiply(0.15);
+
+        // Offset axe backwards
+        Vector backwardOffset = direction.clone()
+            .multiply(-1);
+
+        // Add a downward offset to move the axe lower
+        Vector downwardOffset = new Vector(0, -0.2, 0);
+
+        Location axeLocation = location.clone().add(handOffset).add(backwardOffset).add(downwardOffset);
+
+        ItemDisplay axeDisplay = world.spawn(axeLocation, ItemDisplay.class);
+        axeDisplay.setInvisible(true); // Remove mark in minimaps
+        axeDisplay.setItemStack(item);
+        axeDisplay.setViewRange((float)maxDistance);
```

Comment:
```suggestion
        ItemDisplay axeDisplay = world.spawn(axeLocation, ItemDisplay.class);
        axeDisplay.setInvisible(true); // Remove mark in minimaps
        axeDisplay.setPersistent(false);
        axeDisplay.setItemStack(item);
        axeDisplay.setViewRange((float)maxDistance);
```
We could've avoided the whole detached armor stands problem if the original implementation did this too... But I never really looked closely at Comet's code until now.